host-controller/src/main/java/org/jboss/as/host/controller/ServerInventoryImpl.java

```diff
@@ -401,37 +405,63 @@ public void resumeServer(String serverName) {
     }
 
     @Override
-    public boolean awaitServerSuspend(Set<String> waitForServers, int timeoutInSeconds) {
+    public boolean awaitServerSuspend(Set<String> waitForServers, int timeoutInSeconds, BlockingTimeout blockingTimeout) {
         long end = System.currentTimeMillis() + timeoutInSeconds * 1000;
 
-        Map<String, BlockingQueueOperationListener<?>> listeners = new HashMap<>();
-        //Do one pass initiating the shutdown
+        class OperationData {
+            int blockingTimeout;
+            AsyncFuture<OperationResponse> future;
+            BlockingQueueOperationListener<TransactionalProtocolClient.Operation> listener;
+
+            public OperationData(int blockingTimeout, AsyncFuture<OperationResponse> future, BlockingQueueOperationListener<TransactionalProtocolClient.Operation> listener) {
+                this.blockingTimeout = blockingTimeout;
+                this.future = future;
+                this.listener = listener;
+            }
+        }
+
+        Map<String, OperationData> operationDataMap = new HashMap<>();
         for (String serverName : waitForServers) {
             final ManagedServer server = servers.get(serverName);
             if (server != null) {
                 try {
-                    BlockingQueueOperationListener<?> listener = server.suspend(timeoutInSeconds);
-                    listeners.put(serverName, listener);
+                    int blockingTimeoutValue = blockingTimeout.getProxyBlockingTimeout(server.getAddress(), server.getProxyController());
+                    BlockingQueueOperationListener<TransactionalProtocolClient.Operation> listener = new  BlockingQueueOperationListener<>();
+                    AsyncFuture<OperationResponse> future = server.suspend(timeoutInSeconds, listener);
+
+                    operationDataMap.put(serverName, new OperationData(blockingTimeoutValue, future, listener));
+
                 } catch (IOException e) {
                     HostControllerLogger.ROOT_LOGGER.suspendExecutionFailed(e, serverName);
                 }
             }
         }
 
-        for (Map.Entry<String, BlockingQueueOperationListener<?>> listenerEntry : listeners.entrySet()) {
+        for (Map.Entry<String, OperationData> operationDataEntry : operationDataMap.entrySet()) {
+            final OperationData operationData = operationDataEntry.getValue();
             try {
                 final TransactionalProtocolClient.PreparedOperation<?> prepared =
-                        listenerEntry.getValue().retrievePreparedOperation();
+                        operationData.listener.retrievePreparedOperation(operationData.blockingTimeout, TimeUnit.MILLISECONDS);
+                if ( prepared == null ){
+                    throw new TimeoutException(String.format(
```

Comment:
The catch block L461 will log the TimeoutException, including its message, so if it has a message it has to come from HostControllerLogger so we can have proper i18n. If Kabir's first suggestion doesn't seem good, his second one of a comment and no message in the TimeoutException is fine.

Even if you dropped the @Cause param from the logger methods we'd still want to get rid of the hardcoded exception message. Otherwise at some point someone is sure to come along and decide to "fix" it.